File: statusbar/icon.py
#!/usr/bin/env python

import _thread
import os
import re
import sys
import time

import common

icon_fg = common.pink
icon_bg = common.black
icon_tr = "0xff"
text_fg = common.pink
text_bg = common.black
text_tr = "0xff"

icon_color = "^c" + str(icon_fg) + "^^b" + str(icon_bg) + str(icon_tr) + "^"
text_color = "^c" + str(text_fg) + "^^b" + str(text_bg) + str(text_tr) + "^"
DELAY_TIME = 10

filename = os.path.basename(__file__)
name = re.sub("\\..*", '', filename)


def update(loop=False, set_root=True):
    while True:
        icon = ""
        text = ""
        txt = "^s{}^{} {} {}{} ".format(name, icon_color, icon, text_color, text)
        common.write_to_file(txt + "\n", str(name))
        if not loop:
            if set_root:
                os.system("xsetroot -name '" + str(txt) + "'")
            break
        time.sleep(DELAY_TIME)


def update_thread():
    _thread.start_new_thread(update, (False, False))


def click(string=''):
    match string:
        case 'L':
            os.system("~/.config/rofi/scripts/powermenu_t2")
            pass
        case 'M':
            pass
        case 'R':
            pass
            os.system("feh --randomize --bg-fill ~/Pictures/wallpaper/*.*")
        case 'U':
            pass
        case 'D':
            pass
        case _:
            pass


def notify():
    pass


if __name__ == "__main__":
    if len(sys.argv) > 1:
        if sys.argv[1] == "update":
            pass
        else:
            click(sys.argv[1])
            update(set_root=False)
    else:
        update()

File: statusbar-rs/src/icon.rs
#[cfg(test)]
use regex::Regex;

use crate::common::{Button, cmd, ICON_COLOR, PackageData, TEXT_COLOR};

const NAME: &str = "icon";

pub fn get() -> PackageData {
    let text = format!("^s{}^{} {} {}{} ", NAME, *ICON_COLOR, "", *TEXT_COLOR, "");


    PackageData::new(NAME, text)
}

pub fn api(button: Button) {
    match button {
        Button::LEFT => {
            cmd("~/.config/rofi/scripts/powermenu_t2");
        }
        Button::RIGHT => {
            cmd("feh --randomize --bg-fill ~/Pictures/wallpaper/*.*");
        }
        Button::MIDDLE => {}
        Button::UP => {}
        Button::DOWN => {}
    }
}

#[cfg(test)]
#[test]
pub fn test() {
    println!("get() = {:?}", get());
    let expected_regex = Regex::new(r"\^sicon\^\^c#ff79c6\^\^b#282a360xff\^  \^c#ff79c6\^\^b#282a360xff\^").unwrap();
    assert!(expected_regex.is_match(&get().data));
}